shared/Microsoft.Extensions.StackTrace.Sources/StackFrame/StackTraceHelper.cs

```diff
@@ -119,5 +144,101 @@ internal static MethodDisplayInfo GetMethodDisplayString(MethodBase method)
 
             return methodDisplayInfo;
         }
+
+        private static bool ShowInStackTrace(MethodBase method)
+        {
+            Debug.Assert(method != null);
+
+            // Don't show any methods marked with the StackTraceHiddenAttribute
+            // https://github.com/dotnet/coreclr/pull/14652
+            foreach (var attibute in method.CustomAttributes)
+            {
+                // internal Attribute, match on name
+                if (attibute.AttributeType.Name == "StackTraceHiddenAttribute")
+                {
+                    return false;
+                }
+            }
+
+            var type = method.DeclaringType;
+            if (type == null)
+            {
+                return true;
+            }
+
+            foreach (var attibute in type.CustomAttributes)
+            {
+                // internal Attribute, match on name
+                if (attibute.AttributeType.Name == "StackTraceHiddenAttribute")
+                {
+                    return false;
+                }
+            }
+
+            // Fallbacks for runtime pre-StackTraceHiddenAttribute
+            if (type == typeof(ExceptionDispatchInfo) && method.Name == "Throw")
+            {
+                return false;
+            }
+            else if (type == typeof(TaskAwaiter) ||
+                     type == typeof(TaskAwaiter<>) ||
+                     type == typeof(ConfiguredTaskAwaitable.ConfiguredTaskAwaiter) ||
+                     type == typeof(ConfiguredTaskAwaitable<>.ConfiguredTaskAwaiter))
+            {
+                switch (method.Name)
+                {
+                    case "HandleNonSuccessAndDebuggerNotification":
+                    case "ThrowForNonSuccess":
+                    case "ValidateEnd":
+                    case "GetResult":
```

Comment:
We have a couple of places in Mvc where we manually unwrap tasks using `GetAwaiter().GetResult(). What does the stack trace look like in these cases?

Reply:
Added to test; `Method(string value)` unwraps this way, output is
```
System.Collections.Generic.List<T>+Enumerator.MoveNextRare()
Microsoft.Extensions.Internal.StackTraceHelperTest.Iterator()+MoveNext()
string.Join(string separator, IEnumerable<string> values)
Microsoft.Extensions.Internal.StackTraceHelperTest+GenericClass<T>.GenericMethod<V>(ref V value)
Microsoft.Extensions.Internal.StackTraceHelperTest.MethodAsync(int value)
Microsoft.Extensions.Internal.StackTraceHelperTest.MethodAsync<TValue>(TValue value)
Microsoft.Extensions.Internal.StackTraceHelperTest.Method(string value)
Microsoft.Extensions.Internal.StackTraceHelperTest.StackTraceHelper_ProducesReadableOutput()
```

Reply:
the .ToString() on the Exception is
```
at System.ThrowHelper.ThrowInvalidOperationException_InvalidOperation_EnumFailedVersion()
at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
at Microsoft.Extensions.Internal.StackTraceHelperTest.<Iterator>d__5.MoveNext()
at System.String.Join(String separator, IEnumerable`1 values)
at Microsoft.Extensions.Internal.StackTraceHelperTest.GenericClass`1.GenericMethod[V](V& value)
at Microsoft.Extensions.Internal.StackTraceHelperTest.<MethodAsync>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
at Microsoft.Extensions.Internal.StackTraceHelperTest.<MethodAsync>d__3`1.MoveNext()
--- End of stack trace from previous location where exception was thrown
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
at Microsoft.Extensions.Internal.StackTraceHelperTest.Method(String value)
at Microsoft.Extensions.Internal.StackTraceHelperTest.StackTraceHelper_ProducesReadableOutput()
```